exporter/awsemfexporter/metric_declaration.go

```diff
@@ -34,11 +35,29 @@ type MetricDeclaration struct {
 	// MetricNameSelectors is a list of regex strings to be matched against metric names
 	// to determine which metrics should be included with this metric declaration rule.
 	MetricNameSelectors []string `mapstructure:"metric_name_selectors"`
+	// (Optional) List of label matchers that define matching rules to filter against
+	// the labels of incoming metrics.
+	LabelMatchers []*LabelMatcher `mapstructure:"label_matchers"`
```

Comment:
Make LabelMatchers peer to MetricNameSelectors disable the possibility of handling same metric with different label differently. For example, if customer want some common metric like "go_mem_usage" for only nginx workload but not haproxy.